sql/moz-fx-data-shared-prod/pocket_derived/events_v1/query.py

```diff
@@ -0,0 +1,112 @@
+    #
+    # The Airflow DAG run for 2021-03-01 expects to process complete results up
+    # through 2021-03-01, and executes during UTC day 2021-03-02. That run
+    # will pick up a Pocket result labeled as 2021-03-02, so we need to add an
+    # offset here. Then we make sure to drop partial data for 2021-03-02
+    # in the final query.
+    publish_date = (args.date + timedelta(days=1)).isoformat()
+
+    # First, we load the data from GCS to a temp table.
+    uri = f"gs://{args.source_bucket}/{args.source_prefix}/{publish_date}/*"
+    tmp_suffix = uuid.uuid4().hex[0:6]
+    tmp_table = f"tmp.events_{tmp_suffix}"
+    client.load_table_from_uri(
+        uri,
+        tmp_table,
+        job_config=bigquery.LoadJobConfig(
+            write_disposition=bigquery.job.WriteDisposition.WRITE_TRUNCATE,
+            source_format=bigquery.job.SourceFormat.PARQUET,
+        ),
+    ).result()
+
+    # Next, we run a query that populates the destination table partition.
+    client.query(
+        f"""
+SELECT
+  DATE("{args.date.isoformat()}") AS submission_date,
+  collector_tstamp AS submission_timestamp,
+  STRUCT(
+    STRUCT(GEO_CITY AS city, GEO_COUNTRY AS country) AS geo,
+    STRUCT(
+      CONTEXTS_COM_SNOWPLOWANALYTICS_SNOWPLOW_UA_PARSER_CONTEXT_1[0].useragentFamily AS browser,
+      CONTEXTS_COM_SNOWPLOWANALYTICS_SNOWPLOW_UA_PARSER_CONTEXT_1[0].osFamily AS os,
+      CONCAT(
+        CONTEXTS_COM_SNOWPLOWANALYTICS_SNOWPLOW_UA_PARSER_CONTEXT_1[0].useragentMajor,
+        '.',
+        CONTEXTS_COM_SNOWPLOWANALYTICS_SNOWPLOW_UA_PARSER_CONTEXT_1[0].useragentMinor
+      ) AS version
+    ) AS user_agent
+  ) AS metadata,
+  STRUCT(
+    CONTEXTS_COM_POCKET_USER_1[0].hashed_guid AS app_build,
+    CONTEXTS_COM_POCKET_USER_1[0].hashed_user_id AS client_id,
+    br_lang AS locale,
+    CONTEXTS_COM_SNOWPLOWANALYTICS_SNOWPLOW_UA_PARSER_CONTEXT_1[0].osFamily AS os,
+    CONCAT(
+      CONTEXTS_COM_SNOWPLOWANALYTICS_SNOWPLOW_UA_PARSER_CONTEXT_1[0].osMajor,
+      '.',
+      CONTEXTS_COM_SNOWPLOWANALYTICS_SNOWPLOW_UA_PARSER_CONTEXT_1[0].osMinor
+    ) AS os_version
+  ) AS client_info,
+  CONCAT(EVENT, '.', EVENT_NAME) AS event,
+  derived_tstamp AS event_timestamp,
+  TO_JSON(
+    STRUCT(
+      COALESCE(CONTEXTS_COM_POCKET_API_USER_1[0].api_id, 0) AS api_id,
+      page_urlhost,
+      page_urlpath,
+      page_urlquery,
+      CONTEXTS_COM_POCKET_UI_1[0].type AS type,
+      CONTEXTS_COM_POCKET_UI_1[0].component_detail AS component_detail,
+      CONTEXTS_COM_POCKET_UI_1[0].identifier AS identifier,
+      CONTEXTS_COM_POCKET_UI_1[0].index AS index
+    )
+  ) AS event_extra
+FROM
+  {tmp_table} AS t
+        """,
+        job_config=bigquery.QueryJobConfig(
+            destination=".".join(
+                [
+                    args.project,
+                    args.destination_dataset,
+                    f"{args.destination_table}${args.date.strftime('%Y%m%d')}",
```

Comment:
Do we want to persist older data or do we only care about the most recent import?